src/fsharp/vs/ServiceStructure.fs

```diff
@@ -545,7 +545,9 @@ module Structure =
                     | [], [] -> List.rev res
                     | [], _ -> List.rev (currentBulk::res)
                     | r :: rest, [] -> loop rest res [r]
-                    | r :: rest, last :: _ when r.StartLine = last.EndLine + 1 ->
+                    | r :: rest, last :: _ 
+                        when r.StartLine = last.EndLine + 1 || 
+                             sourceLines.[last.EndLine..r.StartLine - 2] |> Array.forall System.String.IsNullOrEmpty ->
```

Comment:
Are we better off with IsNullOrWhitespace here?

Reply:
omg yes, I was sure I use this function :( Thanks for the review. Fixed.